### PR TITLE
Support new app-exchange APDU

### PR DESCRIPTION
**Pull request by sprohaszka-ledger**
### 📝 Description

Add calls to new APDU coming with new app-exchange version.
Add Moonpay signature.
Update exchange partner signature.
Add unit test and integration test.
Update the `hw-app-exchange` README to ease the use of Speculos.

### ❓ Context

- **Impacted projects**: `hw-app-echange`, `live-common`
- **Linked resource(s)**: [LIVE-9336](https://ledgerhq.atlassian.net/browse/LIVE-9636)

### ✅ Checklist

- [x] **Test coverage** 
- [ ] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-9336]: https://ledgerhq.atlassian.net/browse/LIVE-9336?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/ledgerjs/packages/hw-app-exchange/src/Exchange.ts [hunk 1]
```diff
@@ -12,7 +14,11 @@ export const enum ExchangeTypes {
   Swap = 0x00,
   Sell = 0x01,
   Fund = 0x02,
+  SwapNg = 0x03,
```
Comment (sarneijim) [inline libs/ledgerjs/packages/hw-app-exchange/src/Exchange.ts:17]: What means Ng sufix?

**Review by sarneijim — APPROVED** [review]
LGTM, but my knowledge about this is super limited

File: libs/ledgerjs/packages/hw-app-exchange/src/Exchange.ts (hunk 1 above)
Comment (sprohaszka-ledger) [reply libs/ledgerjs/packages/hw-app-exchange/src/Exchange.ts:17]: It is for Next Gen :P
It comes from the firmware team, for consistency I keep their naming.

**Comment by gre** [comment]
It appears this PR was merged without having a recent enough rebase which caused (due to possible increase of lint strictness?) `develop` to go ❌  since this morning.

Could you fix it asap? Thanks

https://github.com/LedgerHQ/ledger-live/actions/runs/6656746107/job/18089984246

```
@ledgerhq/hw-app-exchange:lint: /home/runner/work/ledger-live/ledger-live/libs/ledgerjs/packages/hw-app-exchange/src/Exchange.integ.test.ts
@ledgerhq/hw-app-exchange:lint:    30:59  error  Replace `·await·appExchangeDatasetTest(⏎······legacySignFormat,⏎····` with `⏎······await·appExchangeDatasetTest(legacySignFormat`  prettier/prettier
@ledgerhq/hw-app-exchange:lint:    67:59  error  Replace `·await·appExchangeDatasetTest(⏎······ngSignFormat,⏎····` with `⏎······await·appExchangeDatasetTest(ngSignFormat`          prettier/prettier
@ledgerhq/hw-app-exchange:lint:   106:59  error  Replace `·await·appExchangeDatasetTest(⏎······ngSignFormat,⏎····` with `⏎······await·appExchangeDatasetTest(ngSignFormat`          prettier/prettier
```

**Pull request merged**